PDF template for model Na 31-2

marlonbaeten opened the pull request:
New Typst template for model Na 31-2

Fixes #1672

Model content:

[Na 31-2 incl. Wpv voorstelPv gsb-cso.pdf](https://github.com/user-attachments/files/20884353/Na.31-2.incl.Wpv.voorstelPv.gsb-cso.pdf)

Model design:

[cso-gsb-pv.pdf](https://github.com/user-attachments/files/20884356/cso-gsb-pv.pdf)

New output:

[model-na-31-2.pdf](https://github.com/user-attachments/files/20884342/model-na-31-2.pdf)

Implementation notes:
- The page "Controles in opdracht van het centraal stembureau" moved to a separate template
- I tried keeping the model itself as simple as possible, styling and scripting is moved to the "common" folder
- The `votes_table` has some complexity related to rows per table/column and balancing columns

You can try out the template in the Typst web environment: https://typst.app/project/pf0ekPUcoIdxIaMpKR83po (@praseodym can give you the correct permissions)

jorisleker reviewed (changes requested):
ssee previous comment

Lionqueen94 commented on backend/templates/common/scripts.typ:
```diff
@@ -242,14 +270,14 @@
 
           if togo > 0 {
-            if calc.rem(togo, half_remainder) == 0 {
+            if calc.rem(togo  +1, half_remainder) == 0 {
```
```suggestion
            if calc.rem(togo + 1, half_remainder) == 0 {
```

Lionqueen94 commented:
We changed the number rendering in Abacus to render each number with the same width, so we want to / are we able to do this in Typst as well? This would make the polling station number list look better I think. @jorisleker

marlonbaeten commented:
> We changed the number rendering in Abacus to render each number with the same width, do we want to / are we able to do this in Typst as well? This would make the polling station number list look better I think. @jorisleker

Yes I'm aware and I already added this. See:

https://github.com/kiesraad/abacus/blob/models-june-2025-typst/backend/templates/common/scripts.typ#L80
https://github.com/kiesraad/abacus/blob/models-june-2025-typst/backend/templates/common/scripts.typ#L272

I will also add this for the polling station numbers :+1:

praseodym commented:
Looking at this version I think it'd be more consistent to always break the candidate votes columns at 25/25/15/15 instead of letting Typst balance the columns. Because the layout for every political group will be the same, I think this will make the model easier to use when filling in by hand (in case of hand counting) and easier to use when doing data entry at the CSB level.

jorisleker commented:
> Looking at this version I think it'd be more consistent to always break the candidate votes columns at 25/25/15/15 instead of letting Typst balance the columns. Because the layout for every political group will be the same, I think this will make the model easier to use when filling in by hand (in case of hand counting) and easier to use when doing data entry at the CSB level.

Agree with this. This will lead to fewer columns and fewer instances where we need subtotals fields. We should not show the subtotals field for <=25 candidates, because then there is only one column, and total == subtotal.

marlonbaeten commented:
> Why was it changed back, I thought rendering `-` was intended ([#1676 (comment)](https://github.com/kiesraad/abacus/pull/1676#issuecomment-3004743760))?

@jorisleker asked me on mattermost to change it back to `0`
